data/msf/sms.rb

```diff
@@ -0,0 +1,186 @@
+          print_error("Exception occured, you done goofed!")
+        ensure
+          http_client.close
+        end
+      end
+
+      def name
+        "sms"
+      end
+
+      def read_settings
+        read = nil
+        if File.exist?(Sms_yaml)
+          ldconfig = YAML.load_file("#{Sms_yaml}")
+          @king_phisher_server = ldconfig['king_phisher_server']
+          @king_phisher_token = ldconfig['king_phisher_token']
+          @sms_number = ldconfig['sms_number']
+          @sms_carrier = ldconfig['sms_carrier']
+          read = true
+        else
+          return read
+        end
+        return read
+      end
+
+      def commands
+        {
+          'sms_start'    => "Start SMS alerts for new sessions",
+          'sms_stop'    => "Stop SMS alerts for new sessions",
+          'sms_save'    => "Save SMS settings to #{Sms_yaml}",
+          'sms_set_server'  => "Set domain name of the King Phisher server",
+          'sms_set_token'    => "Set King Phisher's API token",
+          'sms_set_number'  => "Set number to send SMS alerts to on new session",
+          'sms_set_carrier'  => "Set carrier for sending SMS messages",
+          'sms_show_params'  => "Shows currently set or saved parameters"
+        }
+      end
+
+      def cmd_sms_start
+        if read_settings()
+          self.framework.events.add_session_subscriber(self)
+          print_good("Starting SMS plugin, monitoring sessions...")
+        else
+          print_error("Could not read SMS settings!")
+        end
+      end
+
+      def cmd_sms_stop
+        print_good("Stopping SMS alerting!")
+        self.framework.events.remove_session_subscriber(self)
+      end
+
+      def cmd_sms_save
+        if @king_phisher_server and @king_phisher_token and @sms_number and @sms_carrier
+          config = {
+            'king_phisher_server' => @king_phisher_server,
+            'king_phisher_token' => @king_phisher_token,
+            'sms_number' => @sms_number,
+            'sms_carrier' => @sms_carrier
+          }
+          File.open(Sms_yaml, 'w') do |out|
+            YAML.dump(config, out)
+          end
+          print_good("All parameters saved to #{Sms_yaml}")
+        else
+          print_error("You have not provided all the parameters!")
+        end
+      end
+
+      def cmd_sms_set_server(*args)
+        if args.length > 0
+          print_status("Setting the King Phisher server to #{args[0]}")
+          @king_phisher_server = args[0]
+        else
+          print_error("Please provide the domain name of your King Phisher server!")
+        end
+      end
+
+      def cmd_sms_set_token(*args)
+        if args.length > 0
+          print_status("Setting King Phisher's REST API token to #{args[0]}")
+          @king_phisher_token = args[0]
+        else
+          print_error("Please provide the REST API token of your King Phisher server!")
+        end
+      end
+
+      def cmd_sms_set_number(*args)
+        if args[0].length == 10 
```

Comment:
this should be ```if args[0].to_s.length == 10``` so if args[0] is nil because args.length == 0 it will still work.